pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -214,3 +234,138 @@ func (s *completed) Execute(msg dispatcher.DIDCommMsg) (state, error) {
 		return nil, fmt.Errorf("illegal msg type %s for state %s", msg.Type, s.Name())
 	}
 }
+
+func sendExchangeRequest(msg dispatcher.DIDCommMsg, outbound dispatcher.Outbound, doc *did.Doc) error {
+	requestBytes, dest, err := createExchangeRequest(msg.Payload, doc)
+	if err != nil {
+		return err
+	}
+	//send the exchange request
+	err = outbound.Send(requestBytes, dest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func sendOutbound(msg dispatcher.DIDCommMsg, outbound dispatcher.Outbound) error {
+	if msg.OutboundDestination != nil {
+		dest := &dispatcher.Destination{
+			RecipientKeys:   msg.OutboundDestination.RecipientKeys,
+			ServiceEndpoint: msg.OutboundDestination.ServiceEndpoint,
+			RoutingKeys:     msg.OutboundDestination.RoutingKeys,
+		}
+		err := outbound.Send(msg.Payload, dest)
+		if err != nil {
+			return err
+		}
+	} else{
+		return fmt.Errorf("OutboundDestination cannot be empty for outbound Request")
+	}
+	//Need to figure out how to find the destination for outbound request
```

Comment:
The destination is available in `DIDCommMsg`.

Reply:
DONE